xml_converter/generators/code_generator.py

```diff
@@ -517,42 +533,48 @@ def generate_cpp_variable_data(
-                if fieldval['type'] == "CompoundValue":
-                    xml_export = fieldval['xml_export']
+                if fieldval['type'] in ["CompoundValue", "CompoundCustomClass"]:
                     for component in fieldval['components']:
-                        component_xml_fields = []
-                        component_default_xml_fields = []
-                        for item in component['xml_fields']:
-                            if xml_export == "Children":
-                                component_default_xml_fields.append(item)
-                            if xml_export == "Parent":
-                                component_default_xml_fields.append(fieldval["xml_fields"][0])
-                            component_xml_fields.append(lowercase(item, delimiter=""))
+                        component_xml_fields: List[str] = []
+                        for x in component['xml_fields']:
+                            component_xml_fields.append(lowercase(x, delimiter=""))
+                        component_default_xml_field: str = ""
+                        component_class_name = documentation_type_data[component['type']]["class_name"]
+                        if component['name'] in fieldval['xml_child_export']:
+                            component_default_xml_field = component['xml_fields'][0]
+                            write_to_xml = True
+                        if component['name'] in fieldval['xml_parent_export']:
+                            component_default_xml_field = fieldval['xml_fields'][0]
+                            write_to_xml = False
                         component_attribute_variable = AttributeVariable(
-                            attribute_name=lowercase(component['name'], delimiter="_"),
+                            attribute_name=lowercase(fieldval['name'], delimiter="_") + "." + lowercase(component['name'], delimiter="_"),
                             attribute_type="CompoundValue",
                             cpp_type=doc_type_to_cpp_type[component['type']],
-                            class_name=class_name,
+                            class_name=component_class_name,
                             xml_fields=component_xml_fields,
-                            default_xml_fields=component_default_xml_fields,
-                            xml_export=xml_export,
+                            default_xml_field=component_default_xml_field,
                             protobuf_field=component["protobuf_field"],
-                            compound_name=lowercase(fieldval['name'], delimiter="_"),
+                            parser_flag_name=lowercase(fieldval['name'], delimiter="_"),
+                            write_to_xml=write_to_xml,
+                            is_component=True,
                             args=args,
                         )
                         attribute_variables.append(component_attribute_variable)
+                    if fieldval['xml_parent_export'] == []:
```

Comment:
Add a comment providing some simple logic as to why we want to do this. eg: if we dont have any component fields to export in the parent of a compound (or custom compound) then we dont want to render the parent